demos/http/http_demo_s3_generate_presigned_url/http_demo_s3_generate_presigned_url.c

```diff
@@ -0,0 +1,1206 @@
+    strcat(x_amz_credentials, AWS_S3_BUCKET_REGION);
+    strcat(x_amz_credentials, "/s3/aws4_request");
+    //LogInfo( ( "x_amz_credentials = '%s'", x_amz_credentials ) );
+
+    // https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html
+    char canonical_queries[2048] = "";
+    strcat(canonical_queries, "X-Amz-Algorithm=");
+    strcat(canonical_queries, SIGV4_AWS4_HMAC_SHA256);
+    strcat(canonical_queries, "&X-Amz-Credential=");
+    strcat(canonical_queries, x_amz_credentials);
+    strcat(canonical_queries, "&X-Amz-Date=");
+    strncat(canonical_queries, pDateISO8601, SIGV4_ISO_STRING_LEN);
+    strcat(canonical_queries, "&X-Amz-Expires=3600");
+    strcat(canonical_queries, "&X-Amz-Security-Token=");
+    strncat(canonical_queries, pSecurityToken, securityTokenLen);
+    strcat(canonical_queries, "&X-Amz-SignedHeaders=host");
+
+    /* Setup the HTTP parameters. */
+    sigv4HttpParams.pHttpMethod = requestInfo.pMethod;
+    sigv4HttpParams.httpMethodLen = requestInfo.methodLen;
+    /* None of the requests parameters below are pre-canonicalized */
+    sigv4HttpParams.flags = SIGV4_HTTP_IS_PRESIGNED_URL;
+    sigv4HttpParams.pPath = requestInfo.pPath;
+    sigv4HttpParams.pathLen = requestInfo.pathLen;
+    sigv4HttpParams.pQuery = canonical_queries;
+    sigv4HttpParams.queryLen = strlen(canonical_queries);
+    sigv4HttpParams.pHeaders = pHeaders;
+    sigv4HttpParams.headersLen = headersLen;
+    sigv4HttpParams.pPayload = S3_REQUEST_EMPTY_PAYLOAD;
+    sigv4HttpParams.payloadLen = strlen( S3_REQUEST_EMPTY_PAYLOAD );
+
+    /* Initializing sigv4Params with Http parameters required for the HTTP request. */
+    sigv4Params.pHttpParameters = &sigv4HttpParams;
+
+    if( returnStatus == true )
+    {
+        /* Generate HTTP Authorization header using SigV4_GenerateHTTPAuthorization API. */
+        sigv4Status = SigV4_GenerateHTTPAuthorization( &sigv4Params, pSigv4Auth, &sigv4AuthLen, &signature, &signatureLen );
+
+        if( sigv4Status != SigV4Success )
+        {
+            LogError( ( "Failed to generate HTTP AUTHORIZATION Header. " ) );
+            returnStatus = false;
+        }
+    }
+
+    if( returnStatus == true )
+    {
+        char ota_temp_url[4096] = "https://" AWS_S3_ENDPOINT AWS_S3_URI_PATH "?";
+        strcat(ota_temp_url, "X-Amz-Algorithm=");
+        strcat(ota_temp_url, SIGV4_AWS4_HMAC_SHA256);
+        strcat(ota_temp_url, "&X-Amz-Credential=");
+        size_t encodedLen = sizeof(ota_temp_url) - strlen(ota_temp_url);
+        returnStatus = SigV4_EncodeURI( x_amz_credentials,
+                                        strlen(x_amz_credentials),
+                                        ota_temp_url + strlen(ota_temp_url),
+                                        &encodedLen,
+                                        true/* encode slash */,
+                                        false/* do not double encode equal */ );
+        if( returnStatus != SigV4Success )
```

Comment:
There is a type mismatch here and in a couple other places in this function. `returnStatus` is a bool. You likely meant to use `sigv4Status`